Kzl 241 document controller (#86)
kuzzleio/documentation-V2: src/sdk-reference/cpp/1/document/mUpdate/index.md

---
layout: sdk.html.hbs
algolia: true
title: mUpdate
description:
order: 200
---

# mUpdate

> Updates documents in the persistent data storage.

Reviewer: ```suggestion
Updates multiple documents.
```

> Returns a partial error (with status 206) if one or more documents can not be updated.

Reviewer: ```suggestion
Throws a partial error (error code 206) if one or more documents cannot be updated.
```

> Conflicts may occur if the same document gets updated multiple times within a short time on a database cluster. When this happens, Kuzzle answers with an error that clients have to handle.  

Reviewer: ```suggestion
Conflicts may occur if the same document gets updated multiple times within a short timespan in a database cluster.
```

> You may set the `retryOnConflict` optional argument with a positive integer, asking Kuzzle to retry updating the document that number of times before rejecting the request with an error.

Reviewer: ```suggestion
You can set the `retryOnConflict` optional argument (with a retry count), to tell Kuzzle to retry the failing updates the specified amount of times before rejecting the request with an error.
```
